client/src/helpers/ImagesTreeViewHelpers.js

```diff
@@ -16,20 +16,25 @@ export const DoDataPreprocessing = (filteredActivityMaps, currentExperimentName)
   Object.values(filteredActivityMaps).map(obj => {
     // if experiment key exists
     if (obj.experiment !== undefined) {
-      if (obj.experiment === true) {
+      if(obj.experiment === true){
         // experiment is true
-        obj.hierarchy = obj.hierarchy === undefined || obj.hierarchy.includes(currentExperimentName)
-          ? obj.hierarchy
-          : [currentExperimentName, ...obj.hierarchy];
-      } else {
+        if(obj.hierarchy === undefined){ // no hierarchy key
```

Comment:
This logic and the one above if experiment is false is the same, why the need of the if else then?

Reply:
we have cases where we dont have the experiment key at all where it consider it as experiment = false, and in this case we have to consider it as true, that's why i have made the logic separated in the first place depends if the  experiment exists or not by checking obj.hierarchy === undefined